main/src/main/java/org/mobilitydata/gtfsvalidator/cli/Main.java

```diff
@@ -225,6 +226,14 @@ public static void exportReport(final NoticeContainer noticeContainer, final Arg
       Files.write(
           Paths.get(args.getOutputBase(), args.getValidationReportName()),
           gson.toJson(noticeContainer.exportValidationNotices()).getBytes(StandardCharsets.UTF_8));
+      Files.write(
+          Paths.get(args.getOutputBase(), args.getHtmlReportName()),
+          HtmlOutputUtil.outputBuilder(
+                  noticeContainer.exportValidationNotices(),
```

Comment:
On a similar note, though, I think it would be safer to build the HTML report off of `ValidationReport` instead of the JSON output.

Right now the code is doing this process:
validate->`ValidationReport`->JSON->HTML

The problem with this is that if the JSON representation changes it will break HTML output. And the current `HtmlOutputUtilTest.outputBuilderTest()` would still pass because it's just using a hard-coded representation of the JSON output.

If you take a look at `ValidationReportDeserializer.serialize()`, at the end of the method it builds a `new ValidationReport(noticeReports)` and then converts that to a `JsonObject`.

You should be able to refactor this to effectively build the `ValidationReport` once and then use that as the input to the JSON and HTML output. So the new workflow would look like:
validate->`ValidationReport`->JSON
                                            \->HTML

This would also enable you to build better tests where you could build `Notice`s programmatically and export them to HTML and check the HTML output against that.

Reply:
This have been fixed with the Thymeleaf approach.